www/js/diary/infinite_scroll_list.js

```diff
@@ -167,14 +171,75 @@ angular.module('emission.main.diary.infscrolllist',['ui-leaflet',
         - $ionicScrollDelegate.getScrollPosition().top;
   }
 
-  $scope.readDataFromServer = function() {
+  $scope.populateCompositeTrips = (ctList) => {
+    ctList.forEach((ct, i) => {
+      if ($scope.showPlaces && ct.start_confirmed_place) {
+        const cp = ct.start_confirmed_place;
+        cp.getNextEntry = () => ctList[i];
+        $scope.populateBasicClasses(cp);
+        $scope.labelPopulateFactory.populateInputsAndInferences(cp, $scope.data.manualResultMap);
+        $scope.enbs.populateInputsAndInferences(cp, $scope.data.enbsResultMap);
+      }
+      if ($scope.showPlaces && ct.end_confirmed_place) {
+        const cp = ct.end_confirmed_place;
+        cp.getNextEntry = () => ctList[i + 1];
+        $scope.populateBasicClasses(cp);
+        $scope.labelPopulateFactory.populateInputsAndInferences(cp, $scope.data.manualResultMap);
+        $scope.enbs.populateInputsAndInferences(cp, $scope.data.enbsResultMap);
+        ct.getNextEntry = () => cp;
+      } else {
+        ct.getNextEntry = () => ctList[i + 1];
+      }
+      $scope.populateBasicClasses(ct);
+      $scope.labelPopulateFactory.populateInputsAndInferences(ct, $scope.data.manualResultMap);
+      $scope.enbs.populateInputsAndInferences(ct, $scope.data.enbsResultMap);
+    });
+  }
+
+  $scope.loadWeekOfSelectedDay = () => {
+    const selDay = $scope.infScrollControl.selectedDay;
+    Logger.log("Called loadWeekOfSelectedDay with selectedDay = " + selDay);
+    if (!selDay) return Logger.log("No selected day, early return");
+    const seconds = selDay.getTime() / 1000;
+    const startTs = seconds - (ONE_WEEK / 2);
+    const endTs = seconds + (ONE_WEEK / 2);
+    $scope.infScrollControl.reachedPipelineStart = false;
+    $scope.infScrollControl.reachedPipelineEnd = false;
+    $scope.data.allTrips = [];
+    $scope.data.listEntries = [];
+    $scope.readDataFromServer(startTs, endTs);
```

Comment:
Is this the right behavior to expect?
I did think over this for a while - and it seems fine to me.

The common use case: If you're looking at your recent trips, and you want to jump to a particular date in the past, you don't want to see the trips you were just looking at. You only want to see what you asked for - the newly loaded trips.
Then from there, you could load the next/previous week if you wanted.

I think it would be weird if the original trips stayed in the view when you asked for a specific date. (What would we do if there's a big gap in between?)

Reply:
This behavior is fine. It was just to keep this clear in my head 😄